cmd/worker/worker.go

```diff
@@ -186,25 +187,20 @@ func (c *Command) Start(ctx context.Context) error {
 
 	certManager := worker.NewCertificateManager(ctx, kubeletKubeconfigPath)
 	if !c.SingleNode && !c.EnableWorker {
-		clusterConfig, err := config.LoadClusterConfig(c.K0sVars)
-		if err != nil {
-			return fmt.Errorf("failed to load cluster config: %w", err)
-		}
-
 		componentManager.Add(ctx, &status.Status{
 			Prober: prober.DefaultProber,
 			StatusInformation: status.K0sStatus{
-				Pid:           os.Getpid(),
-				Role:          "worker",
-				Args:          os.Args,
-				Version:       build.Version,
-				Workloads:     true,
-				SingleNode:    false,
-				K0sVars:       c.K0sVars,
-				ClusterConfig: clusterConfig,
+				Pid:             os.Getpid(),
+				Role:            "worker",
+				Args:            os.Args,
+				Version:         build.Version,
+				Workloads:       true,
+				SingleNode:      false,
+				K0sVars:         c.K0sVars,
+				BootstrapConfig: nil,
```

Comment:
worker has no local config

Reply:
also i think it needs to use the workerside kubeclient thing for fetching the dynamic config since the regular one uses adminkubeconfig

Reply:
trying to use the autopilot factory to make client now